components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2792,10 +2792,25 @@ private void processPatchRequest(List<Patch> patchRequest, IdentityProvider idpT
 
     private void patchIdpProperties(IdentityProvider identityProvider, String propertyName, String propertyValue) {
 
+        Boolean hasExistingJWKSUri = false;
         IdentityProviderProperty[] propertyDTOS = identityProvider.getIdpProperties();
         for (IdentityProviderProperty propertyDTO : propertyDTOS) {
             if (propertyName.equals(propertyDTO.getName())) {
                 propertyDTO.setValue(propertyValue);
+                if (propertyName.equals(Constants.JWKS_URI)) {
+                    hasExistingJWKSUri = true;
+                }
+            }
+        }
+        if (!hasExistingJWKSUri && propertyName.equals(Constants.JWKS_URI)) {
```

Comment:
Change this as `Constants.JWKS_URI.equals(propertyName)` to avoid possible NPEs.